lib/if.c

```diff
@@ -1060,11 +1060,21 @@ void if_terminate(struct vrf *vrf)
 	while (!RB_EMPTY(if_name_head, &vrf->ifaces_by_name)) {
 		ifp = RB_ROOT(if_name_head, &vrf->ifaces_by_name);
 
-		if (ifp->node) {
-			ifp->node->info = NULL;
-			route_unlock_node(ifp->node);
-		}
-		if_delete(&ifp);
+		/*
+		 * If the default VRF is terminated it means that
+		 * the program is shutting down and we need to
+		 * delete all the interfaces.
+		 * If any other VRF is terminated, we only need
+		 * to move its interfaces to the default VRF.
+		 */
+		if (vrf->vrf_id == VRF_DEFAULT) {
+			if (ifp->node) {
+				ifp->node->info = NULL;
+				route_unlock_node(ifp->node);
+			}
+			if_delete(&ifp);
+		} else
+			if_update_to_new_vrf(ifp, VRF_DEFAULT);
```

Comment:
Per style guide this `else` should be in `{}` also

Reply:
Fixed and opened a backport PR https://github.com/FRRouting/frr/pull/8439